src/figuresClasses.ts

```diff
@@ -2,10 +1,85 @@
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('all sides of triangle must be greater than 0');
+    }
+
+    const biggestSide = Math.max(a, b, c);
+    const sumOfSmaller = a + b + c - biggestSide;
+
+    if (biggestSide >= sumOfSmaller) {
+      throw new Error(`sides ${a}, ${b} and ${c} can not form a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const semiperimeter = 0.5 * (this.a + this.b + this.c);
 
-export class Triangle implements Figure {}
+    return (
+      Math.floor(
+        Math.sqrt(
+          semiperimeter *
+            (semiperimeter - this.a) *
+            (semiperimeter - this.b) *
+            (semiperimeter - this.c),
+        ) * 100,
+      ) / 100
+    );
+  }
+}
+
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('radius of circle must be greater than 0');
+    }
+  }
+
+  getArea(): number {
```

Comment:
The `getArea` function should not accept any parameters as it can access `radius` directly from the object's fields.